app/controllers/users/sessions_controller.rb

```diff
@@ -39,6 +40,13 @@ def timeout
       redirect_to root_url
     end
 
+    protected
+
+    def require_no_authentication
```

Comment:
It does get called by default in the `Devise::SessionController`. I think we need to override this method because the `after_sign_in_path_for` method that we've overridden in the `ApplicationController`  tries to redirect the user to /profile, which in turn redirects the user back to whatever 2fa screen they are currently on.